### PR TITLE
Added C++ implementation of bisection method

### LINKED ISSUE #1914: Bisection method in cpp
I would like to add a solution in cpp to implement root finding method i.e The Bisection method.Please assign me this issue

### PR DESCRIPTION
**Pull request by Nikhil-Patro**
Fixes #1914

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added a C++ implementation of the bisection method to find roots of polynomials.
#### Languages Used:
- C++

#### Files Added:
- Bisection_Method.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Nikhil-Patro** [comment]
This is the new PR with all changes implemented and multiple commits squashed.Please review @asha15 .

**Comment by Nikhil-Patro** [comment]
Please review @asha15

File: Bisection_Method/Bisection_Method.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    if(polynomial(lower_bound) * polynomial(upper_bound) > 0)
+    {
+        cout << "Invalid interval" << endl;
+    }
+    else
+    {   // repeat until the required accuracy is obtained
+        while(abs(polynomial(mid(lower_bound, upper_bound))) > error)
+        {
+            cout << "\nIteration_no = " << i << " ,First_no = " << lower_bound << " ,Second_no = " << upper_bound << " middle = " << mid(lower_bound, upper_bound) ;
+            cout << " ,f(middle) = " << polynomial(mid(lower_bound, upper_bound)) << " ,f(first_no)*f(middle) = " << polynomial(lower_bound)*polynomial(mid(lower_bound, upper_bound)) << endl;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: one line should contain max 80 characters. please break this lines

File: Bisection_Method/Bisection_Method.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            }
+            else
+            {
+                lower_bound = mid(lower_bound, upper_bound);
+            }
+            i++;
+        }
+        cout << "\nIteration_no = " << i << " ,First_no = " << lower_bound << " ,Second_no = " << upper_bound << " middle = " << mid(lower_bound, upper_bound) ;
+        cout << " ,f(middle) = " << polynomial(mid(lower_bound, upper_bound)) << " ,f(first_no)*f(middle) = " << polynomial(lower_bound)*polynomial(mid(lower_bound, upper_bound)) << endl;
+        cout << "\nThe approximated root of the given polynomial within the specified interval (" << a << " , " << b << ") = " ;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: one line should contain max 80 characters. please break this lines

File: Bisection_Method/Bisection_Method.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        cout << mid(upper_bound, lower_bound) << endl;
+    }
+    return 0;
+}
+
+/* OUTPUT ===============================================================================================================
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: remove the equal signs

**Comment by somya-kapoor** [comment]
@Nikhil-Patro Are you working on it?

**Comment by Nikhil-Patro** [comment]
> @Nikhil-Patro Are you working on it?

yes , i have made new PR for these changes

**Comment by Nikhil-Patro** [comment]
I have made new PR review there

**Pull request closed without merging**